### PR TITLE
sig-pm: Retire/delete/transfer SIG's GitHub teams

### PR DESCRIPTION
**Pull request by justaugustus**
(Part of https://github.com/kubernetes/community/issues/4721.)

- sig-pm: Retire/delete/transfer SIG's GitHub teams
  To SIG Arch:
  - `enhancements-admins`
  - `enhancements-maintainers`
  
  To SIG ContribEx:
  - `project-board-maintainers`
- sig-arch: Update Enhancements subproject owners (cc: @jeremyrickard)

For SIG Arch:
/assign @johnbelamaric @dims @derekwaynecarr 

For SIG ContribEx:
/assign @mrbobbytables @castrojo

### CODE REVIEW

**Comment by mrbobbytables** [comment]
/lgtm
from contribex side of things 👍

File: config/kubernetes/sig-pm/OWNERS [hunk 1]
```diff
@@ -1,8 +0,0 @@
-# See the OWNERS docs at https://go.k8s.io/owners
-
-reviewers:
-  - sig-pm-leads
```
Comment (nikhita) [inline config/kubernetes/sig-pm/OWNERS:4]: Can you also remove this alias from OWNERS_ALIASES?

File: config/kubernetes/sig-contributor-experience/teams.yaml [hunk 2]
```diff
@@ -47,6 +47,18 @@ teams:
     - Phillels
     - tpepper
     privacy: closed
+  project-board-maintainers:
```
Comment (nikhita) [inline config/kubernetes/sig-contributor-experience/teams.yaml:50]: Out of curiosity, is this team being used anywhere?

Comment (nikhita) [reply config/kubernetes/sig-contributor-experience/teams.yaml:50]: Never mind, figured it out: https://github.com/orgs/kubernetes/teams/project-board-maintainers/projects :)

File: config/kubernetes/sig-pm/OWNERS (hunk 1 above)
Comment (justaugustus) [reply config/kubernetes/sig-pm/OWNERS:4]: @nikhita -- Removed! :)

**Comment by johnbelamaric** [comment]
/approve
for sig-arch

**Comment by mrbobbytables** [comment]
oh derp, needed lgtm too.
Taking into account john's approve and my earlier lgtm..just gonna go ahead :P
/lgtm

**Pull request merged**